pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -312,50 +392,18 @@ func (e *DockercfgController) createTokenSecret(serviceAccount *api.ServiceAccou
 			Annotations: map[string]string{
 				api.ServiceAccountNameKey: serviceAccount.Name,
 				api.ServiceAccountUIDKey:  string(serviceAccount.UID),
+				PendingDockercfgNameKey:   "true",
```

Comment:
@liggitt agree, I don't think we need any annotation as we can just check if the state is pending by looking into secret data map.